odoo/addons/base/tests/common.py

```diff
@@ -106,11 +106,11 @@ def _use_currency(cls, company, currency_code):
         # Enforce constant currency
         currency = cls._enable_currency(currency_code)
         if company.currency_id != currency:
-            cls.env.transaction.cache.set(cls.env.company, type(cls.env.company).currency_id, currency.id, dirty=True)
             # this is equivalent to cls.env.company.currency_id = currency but without triggering buisness code checks.
             # The value is added in cache, and the cache value is set as dirty so that that
             # the value will be written to the database on next flush.
             # this was needed because some journal entries may exist when running tests, especially l10n demo data.
+            company._update_cache({'currency_id': currency.id})
```

Comment:
That's not the same. _update_cache doesn't set the dirty `flag=True` :thinking: ? the _update_cache API is very confusing IMO.

Reply:
Change removed.